### PR TITLE
0.12.0-beta Bug Fixes

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Rivet/Functions/Get-MigrationFile.ps1 [hunk 1]
```diff
@@ -108,14 +104,22 @@ function Get-MigrationFile
             }
 
             $migration = $_
-            $Exclude | Where-Object { $migration.MigrationID -notlike $_ -and $migration.MigrationName -notlike $_ -and $migration.BaseName -notlike $_ }
+            foreach( $pattern in $Exclude )
+            {
+                if( $migration.MigrationID -like $pattern -or $migration.MigrationName -like $pattern -or $migration.BaseName -like $pattern )
```
Comment (pnstickne) [inline Rivet/Functions/Get-MigrationFile.ps1]: Each or on own line would make for easier vertical reading.

File: Source/Operations/RenameObjectOperation.cs [hunk 2]
```diff
@@ -16,6 +16,22 @@ public class RenameObjectOperation : RenameOperation
...
 		public override string ObjectName => $"{SchemaName}.{Name}";
 
+		protected override MergeResult DoMerge(Operation operation)
+		{
+			if (base.DoMerge(operation) == MergeResult.Stop)
+				return MergeResult.Stop;
+
+			if (operation is RenameTableObjectOperation otherAsRenameTableObjectOp &&
+			    ObjectName.Equals(otherAsRenameTableObjectOp.TableObjectName, StringComparison.InvariantCultureIgnoreCase))
+
```
Comment (pnstickne) [inline Source/Operations/RenameObjectOperation.cs]: Extra newline.

File: Source/Operations/RenameTableObjectOperation.cs [hunk 3]
```diff
@@ -32,7 +32,11 @@ protected override MergeResult DoMerge(Operation operation)
                 TableObjectName.Equals(otherAsRenameOp.ObjectName, StringComparison.InvariantCultureIgnoreCase))
 
             {
-                TableName = otherAsRenameOp.NewName;
+	            otherAsRenameOp.OnDisabled += (sender, args) =>
+	            {
+		            if (sender is RenameOperation disabledRenameTableOp) 
+			            TableName = disabledRenameTableOp.NewName;
+	            };
                 return MergeResult.Continue;
```
Comment (pnstickne) [inline Source/Operations/RenameTableObjectOperation.cs:40]: Indent.

File: Source/Operations/AddDefaultConstraintOperation.cs [hunk 4]
```diff
@@ -22,7 +22,8 @@ public sealed class AddDefaultConstraintOperation : ConstraintOperation
 
 		public override string ToIdempotentQuery()
 		{
-			return $"if object_id('{SchemaName}.{Name}', 'D') is null{Environment.NewLine}\t{ToQuery()}";
+			return $"if object_id('{SchemaName}.{Name}', 'D') is null{Environment.NewLine}" +
```
Comment (pnstickne) [inline Source/Operations/AddDefaultConstraintOperation.cs:25]: Other similar constructs use BEGIN/END; adding such might increase uniformity.

**Review by pnstickne — DISMISSED** [review]

File: Source/Operations/AddDefaultConstraintOperation.cs (hunk 4 above)
Comment (splatteredbits) [reply Source/Operations/AddDefaultConstraintOperation.cs:25]: I only use BEGIN/END in situations where the query that executes when the conditional is true spans multiple lines.

**Comment by splatteredbits** [comment]
OK. Ready for next review.

File: Rivet/Functions/Operations/Remove-DefaultConstraint.ps1 [hunk 5]
```diff
@@ -43,7 +43,15 @@ function Remove-DefaultConstraint
 
     if( -not $ColumnName )
     {
-        Write-Warning -Message ('Not providing the ColumnName parameter is obsolete. This parameter will be mandatory in a future version of Rivet. Please pass the column name whose default constraint is being removed to the ColumnName parameter.')
+        $nameMsg = ''
+        if( $Name )
+        {
+            $nameMsg = "'s $($Name) constraint"
+        }
+        $msg = ('The ColumnName parameter will be required in a future version of Rivet. Add a ""ColumnName"" ' +
```
Comment (pnstickne) [inline Rivet/Functions/Operations/Remove-DefaultConstraint.ps1:51]: `'..""..'` Results in extra quotes in message.

While a preference, usage might be more consistent as:

    "..`".."

File: Source/Operations/AddStoredProcedureOperation.cs [hunk 6]
```diff
@@ -15,13 +15,14 @@ public AddStoredProcedureOperation(string schemaName, string name, string defini
 
 		public override string ToIdempotentQuery()
 		{
-			return string.Format("if object_id('{0}.{1}', 'P') is null and object_id('{0}.{1}', 'PC') is null{2}\texec sp_executesql N'{3}'", SchemaName, Name,
-				Environment.NewLine, ToQuery().Replace("'", "''"));
+			return
```
Comment (pnstickne) [inline Source/Operations/AddStoredProcedureOperation.cs:18]: Line break feels odd.

File: Source/Operations/Operation.cs [hunk 7]
```diff
@@ -90,6 +106,11 @@ protected virtual MergeResult DoMerge(Operation operation)
 			return MergeResult.Continue;
 		}
 
+		protected string ToIndentedQuery()
```
Comment (pnstickne) [inline Source/Operations/Operation.cs:109]: I might structure as `AsFormattedQuery(ToQuery());` in usages to minimize hidden virtual method call.

Also, perhaps always add a newline if required?

**Review by pnstickne — APPROVED** [review]

File: Source/Operations/AddStoredProcedureOperation.cs (hunk 6 above)
Comment (splatteredbits) [reply Source/Operations/AddStoredProcedureOperation.cs:18]: ReSharper.

**Pull request merged**